utils/cached_file.hh

```diff
@@ -208,26 +218,84 @@ public:
     class page_view {
         cached_page::ptr_type _page;
         size_t _offset;
-        size_t _size;
+        size_t _size = 0;
         std::optional<reader_permit::resource_units> _units;
     public:
         page_view() = default;
+
         page_view(size_t offset, size_t size, cached_page::ptr_type page, std::optional<reader_permit::resource_units> units)
                 : _page(std::move(page))
                 , _offset(offset)
                 , _size(size)
                 , _units(std::move(units))
         {}
 
+        page_view(page_view&& o) noexcept
+            : _page(std::move(o._page))
+            , _offset(std::exchange(o._offset, 0))
+            , _size(std::exchange(o._size, 0))
+            , _units(std::move(o._units))
+        {}
+
+        page_view& operator=(page_view&& o) noexcept {
+            _page = std::move(o._page);
+            _offset = std::exchange(o._offset, 0);
+            _size = std::exchange(o._size, 0);
+            _units = std::move(o._units);
+            return *this;
+        }
+
         // The returned buffer is valid only until the LSA region associated with cached_file invalidates references.
         temporary_buffer<char> get_buf() {
             auto buf = _page->get_buf_weak();
-            buf.trim(_size);
             buf.trim_front(_offset);
+            buf.trim(_size);
             return buf;
         }
```

Comment:
`page_view::get_buf()` and `cached_page::get_buf_weak()` are unused after your series. And we don't want to keep them, because they are unsafe. Remove them.

Reply:
Yes.